plugin/pkg/scheduler/algorithm/priorities/selector_spreading.go

```diff
@@ -206,6 +206,21 @@ func NewServiceAntiAffinityPriority(podLister algorithm.PodLister, serviceLister
 	return antiAffinity.CalculateAntiAffinityPriority
 }
 
+// Classifies nodes into ones with labels and without labels.
+func (s *ServiceAntiAffinity) getNodeClassificationByLabels(nodes []*v1.Node) (map[string]string, []string) {
+	labeledNodes := map[string]string{}
+	nonLabeledNodes := []string{}
+	for _, node := range nodes {
```

Comment:
fwiw as a follow on... I think we need to start creating cache indexes, b/c we're doing a bunch of operations sub-optimally.